### PR TITLE
Update SQL REST API version to 2021-11-01

### LINKED ISSUE #114: verify REST API client version
see if Azure resource manager client has an update, validate that we're not at risk of being on a deprecated version before releasing v2

https://github.com/Azure/sql-action/blob/7e69fdc44aba3f05fd02a6a4190841020d9ca6f7/src/AzureSqlResourceManager.ts#L3

### PR DESCRIPTION
**Pull request by zijchen**
Fixes #114 

The 2014 API version is to be retired in September, replaced by a stable release of the 2021-11-01 version. 

API mappings from 2014 to 2021 can be found here: https://review.docs.microsoft.com/en-us/rest/api/sql/retirement?branch=sqlretirement

Luckily for us, the definitions of the 3 API calls we use aren't changed besides the version.

### CODE REVIEW

**Comment by zijchen** [comment]
Note this PR is to the v2 branch. 2014 version should continue to work after September but if it doesn't, we can cherry pick this into master branch/v1 releases as well.

File: src/AzureSqlResourceManager.ts [hunk 1]
```diff
@@ -2,6 +2,9 @@ import { IAuthorizer } from 'azure-actions-webclient/Authorizer/IAuthorizer';
 import { WebRequest } from 'azure-actions-webclient/WebClient';
 import { ServiceClient as AzureRestClient, ToError, AzureError } from 'azure-actions-webclient/AzureRestClient'
 
+// API docs: https://docs.microsoft.com/rest/api/sql
+const SqlApiVersion = '2021-11-01';
```
Comment (chlafreniere) [inline src/AzureSqlResourceManager.ts:6]: nit: do we have a convention for constants in this repo? e.g. SQL_API_VERSION?

**Review by chlafreniere — APPROVED** [review]
Nice work, @zijchen!

**Review by kisantia — APPROVED** [review]

File: src/AzureSqlResourceManager.ts (hunk 1 above)
Comment (zijchen) [reply src/AzureSqlResourceManager.ts:6]: No convention I'm aware of, although we should probably have a conversation about that in the future as we implement more features!

**Pull request merged**